File: CloudFormation/src/test/java/com/intellij/aws/cloudformation/tests/RenameTests.java
package com.intellij.aws.cloudformation.tests;

import com.intellij.codeInsight.TargetElementUtilBase;
import com.intellij.psi.PsiElement;
import com.intellij.refactoring.rename.RenameProcessor;
import com.intellij.testFramework.LightCodeInsightTestCase;
import org.jetbrains.annotations.NotNull;

public class RenameTests extends LightCodeInsightTestCase {
  public void testSimpleEntity() throws Exception {
    configureByFile("simpleEntity.template");
    PsiElement element = TargetElementUtilBase.findTargetElement(
        myEditor, TargetElementUtilBase.ELEMENT_NAME_ACCEPTED | TargetElementUtilBase.REFERENCED_ELEMENT_ACCEPTED);
    new RenameProcessor(getProject(), element, "NEW_NAME", false, false).run();
    checkResultByFile("simpleEntity.after.template");
  }

  @NotNull
  @Override
  protected String getTestDataPath() {
    return TestUtil.getTestDataPath("rename");
  }
}

File: CloudFormation/src/test/kotlin/com/intellij/aws/cloudformation/tests/RenameTests.kt
package com.intellij.aws.cloudformation.tests

import com.intellij.codeInsight.TargetElementUtil
import com.intellij.refactoring.rename.RenameProcessor
import com.intellij.testFramework.LightPlatformCodeInsightTestCase

class RenameTests : LightPlatformCodeInsightTestCase() {
  fun testSimpleEntity() {
    configureByFile("simpleEntity.template")
    val element = TargetElementUtil.findTargetElement(
        editor, TargetElementUtil.ELEMENT_NAME_ACCEPTED or TargetElementUtil.REFERENCED_ELEMENT_ACCEPTED)!!
    RenameProcessor(getProject(), element, "NEW_NAME", false, false).run()
    checkResultByFile("simpleEntity.after.template")
  }

  override fun getTestDataPath(): String {
    return TestUtil.getTestDataPath("rename")
  }
}
